torch/_subclasses/functional_tensor.py

```diff
@@ -520,7 +520,25 @@ def inner(*args, **kwargs):
         disable_above = torch._C._ExcludeDispatchKeyGuard(
             torch._C.DispatchKeySet(torch._C.DispatchKey.Functionalize)
         )
-        with disable_above, mode:
+
+        # FIXME (tmanlaibaatar) This is little weird.
+        # when map_impl.autograd is invoked, we shouldn't be
+        # tracing through predispach mode stack. But since this mode is
+        # saved before this happens on the PythonFunctinalizeAPI object,
+        # it will still try to dispatch through PreDispach FunctionalTensorMode
+        # This is wrong because at this time, PreDispatch mode stack is already
+        # not active. So we hack it to manually dispatch to aot-dispatch FunctionalTensorMode
+        # if we figure there is no PreDispatch mode active. Seems very hacky??
```

Comment:
err, do you have a concrete example of what is going on

> But since this mode is saved

What does this mean?